### ISSUE TITLE
Prevent Variable Translation

### ISSUE THREAD

**Issue by sitranok** [dolanmiu/grunt-google-translate#3]
My apologies if this isn't the right place to ask this.  I'm using this plugin to generate angular-translate translation files for various languages and my original file has some text strings that include placeholders for variable replacement with angular-translate and the output I'm getting back from Google for some of the languages is translating the variable placeholder.

For example I've got one string that's "Choose between {{minimum}} and {{maximum}}" and in the Spanish file that's generated it ends up being "Elija entre {{}} mínimo y máximo {{}}".

Does anyone know of a pattern I could follow that would prevent those placeholders from being replaced in translation?

**Comment by dolanmiu** [comment]
Hi, yes I am aware of this issue
I have this problem too actually (I am the developer)

At the moment, I crudely replace the variables after translating every single time. Not ideal I know, just need to get round to fixing it

This plugin uses the popular google-translate NPM at the core

I've got an idea of replacing the variables like {{ minimum }} into: {{ VAR_1 }} before sending it off to translation, then replacing it back after translation.

What do you think?

Since you asked for this, I am motivated to fix it :) I will get it done in a few days

**Comment by dolanmiu** [comment]
If you can fix it, then even better :)

**Comment by sitranok** [comment]
I had actually implemented a similar approach, though using grunt-replace and a separate json file containing the replacement variables and original value.  I had that approach in place already to handle things like brand names that I didn't want translated and just hacked the angular variables in there as a stopgap.  Having that built in would make things a lot cleaner going forward, though I don't know if I will have time in the next few days.

If you do decide to go forward with implementing that yourself one thing I came across was that even if you replace the contents of the interpolation bindings with something untranslatable google translate will still sometimes mangle them.  I've seen things like "{{}}minimum" and "{{and mimimum}}" in the destination files that the replace didn't handle properly.  I ended up replacing the entire binding with a variable "VAR_1" and then replacing that with "{{ minimum }}" and haven't run into any issues with that approach yet.
